getGeocode.js

```diff
@@ -50,13 +52,18 @@ const getAllCoordinates = async (locations, cachedResults) => {
                 return { ...location, ...existingLocations[locationInd] };
             } else {
                 const locationData = await getGeocode(name, street, zip);
-                await sleep(100);
-                return {
-                    ...location,
-                    resolvedLocation: locationData.results[0].formatted_address,
-                    latitude: locationData.results[0].geometry.location.lat,
-                    longitude: locationData.results[0].geometry.location.lng,
-                };
+
+                if (locationData) {
+                    return {
+                        ...location,
+                        resolvedLocation:
+                            locationData?.results[0].formatted_address,
+                        latitude:
+                            locationData?.results[0].geometry.location.lat,
+                        longitude:
+                            locationData?.results[0].geometry.location.lng,
+                    };
+                } else return location;
```

Comment:
In the case that the geocode fails with 5 delayed retries, the function will just return the location without the geocode data.